src/VsixTesting.Installer/Program.cs

```diff
@@ -649,4 +665,116 @@ internal static class AppDomainExtensions
         public static T CreateInstanceFromAndUnwrap<T>(this AppDomain domain)
             => (T)domain.CreateInstanceFromAndUnwrap(typeof(T).Assembly.Location, typeof(T).FullName);
     }
+
+    internal class VisualStudioUtil
+    {
+        public const string ProcessName = "devenv";
+
+        public static DTE GetDTE(Process process)
+        {
+            string namePattern = $@"!VisualStudio\.DTE\.(\d+\.\d+):{process.Id.ToString()}";
+            return (DTE)RunningObjectTable.GetRunningObjects(namePattern).FirstOrDefault();
+        }
+
+        public static IEnumerable<DTE> GetRunningDTEs()
+        {
+            foreach (var process in Process.GetProcessesByName(ProcessName))
+            {
+                if (GetDTE(process) is DTE dte)
+                    yield return dte;
+            }
+        }
+
+        public static DTE GetDteFromDebuggedProcess(Process process)
+        {
+            foreach (var dte in GetRunningDTEs())
+            {
+                try
+                {
+                    if (dte.Debugger.CurrentMode != EnvDTE.dbgDebugMode.dbgDesignMode)
+                    {
+                        foreach (Process2 debuggedProcess in dte.Debugger.DebuggedProcesses)
+                        {
+                            if (debuggedProcess.ProcessID == process.Id)
+                                return dte;
+                        }
+                    }
+                }
+                catch (COMException)
+                {
+                    continue;
```

Comment:
Could you add `Trace.WriteLine(e)`, so I could use DebugView to see if something is going wrong?